rpcHub/rpcHub.go

```diff
@@ -113,6 +113,8 @@ func dialRpcClient(servAddr string, portNo uint16) (*rpc.Client, net.Conn) {
 // Get a client to the rpc server
 func Client(servAddr string, portNo uint16) *RpcClient {
 	clientKey := fmt.Sprintf("%s:%d", servAddr, portNo)
+	dbLock.Lock()
+	defer dbLock.Unlock()
```

Comment:
It required a RLock() for reading it followed by a write lock to store new client.
I thought i'll simplify the code by taking a full lock at the top..